spec/models/cart_spec.rb

```diff
@@ -136,5 +136,27 @@
       expect(email_recipients).to eq(['requester@some-dot-gov.gov'])
     end
   end
+  describe '#restart' do
+    it 'resets approval states when rejected' do
+      cart = FactoryGirl.create(:cart_with_approvals)
+      Dispatcher.deliver_new_cart_emails(cart)
+      expect(cart.api_tokens.length).to eq(2)
+
+      cart.approver_approvals.first.approve!
+      cart.approver_approvals.last.reject!
+      expect(cart.approvals.where(status: 'approved').length).to eq(1)
+      expect(cart.approvals.where(status: 'rejected').length).to eq(1)
```

Comment:
Hmm, wonder if we can move some of this setup into a factory or something. Not a blocker.